### PR TITLE
libcurl: conan support mbedtls

### PR DESCRIPTION
**Pull request by HypoYoung**
Specify library name and version:  **libcurl/7.84.0**

libcurl  can support mbedtls, and mbedtls is a lib that is used in many softwares so I modify the conan recipies of libcurl to support mbedtls.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -584,6 +592,10 @@ def _generate_with_cmake(self):
             tc.variables["CURL_USE_WOLFSSL"] = self.options.with_ssl == "wolfssl"
         else:
             tc.variables["CMAKE_USE_WOLFSSL"] = self.options.with_ssl == "wolfssl"
+        if Version(self.version) >= "7.81.0":
+            tc.variables["CIRL_USE_MBEDTLS"] = self.options.with_ssl == "mbedtls"
```
Comment (jwillikers) [inline recipes/libcurl/all/conanfile.py:596]: typo?

```suggestion
            tc.variables["CURL_USE_MBEDTLS"] = self.options.with_ssl == "mbedtls"
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (HypoYoung) [reply recipes/libcurl/all/conanfile.py:596]: thanks for review, it' s a mistake, shoud I restart a new pr?

Comment (jwillikers) [reply recipes/libcurl/all/conanfile.py:596]: No, you can just update this one and CI should automatically retrigger. Thanks!

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by jwillikers** [comment]
@HypoYoung I think you need to sign the CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=20639)

**Comment by HypoYoung** [comment]
> @HypoYoung I think you need to sign the CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=20639)
Thanks, I've signed CLA. Is there anything that I shoud do next?

**Comment by jwillikers** [comment]
> > @HypoYoung I think you need to sign the CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=20639)
> > Thanks, I've signed CLA. Is there anything that I shoud do next?

No, we'll just wait for someone from the Conan team to fix the missing dependencies and review.

**Comment by uilianries** [comment]
Sorry the delay, I just pulled the master branch over your PR, so it will re-trigger the entire build.

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by HypoYoung** [comment]
> LGTM
I can not see the error information during Conan v2 build, how can I see the error information?

**Comment by uilianries** [comment]
> Sorry, the build is only launched for Access Request users. You can request access https://github.com/conan-io/conan-center-index/issues/4.

@HypoYoung Please, ask for access request in the issue #4. The CI build will not allow to merge your PR in case you are not listed as allowed contributor. Regards!

**Review by SpaceIm — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**